src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -724,12 +725,22 @@ protected override bool EnsureIncludeSerializer()
         protected override bool EnsureIncludeDeserializer()
         {
             // TODO -- this should always return true when use model reader writer is enabled.
-            return Configuration.UseModelReaderWriter || _usage.HasFlag(InputModelTypeUsage.Output);
+            return !MgmtReferenceType.IsReferenceType(InputModel) && (Configuration.UseModelReaderWriter || _usage.HasFlag(InputModelTypeUsage.Output));
         }
 
         protected override JsonObjectSerialization? BuildJsonSerialization()
         {
-            return MgmtReferenceType.IsReferenceType(InputModel) || !InputModel.Serialization.Json ? null : _serializationBuilder.BuildJsonObjectSerialization(InputModel, this);
+            return !InputModel.Serialization.Json ? null : _serializationBuilder.BuildJsonObjectSerialization(InputModel, this);
+        }
+
+        protected override ObjectTypeSerialization BuildSerialization()
+        {
+            bool isReference = MgmtReferenceType.IsReferenceType(InputModel);
```

Comment:
what are we doing here for refenrenceType?

Reply:
reference types don't have any interface. it is our existing behavior.

Reply:
My question is why do we add a parameterless constructor of `SerializationInterfaces` and use it for referenceType only here?

Reply:
What we expect is:
Normal model: has interface, has writecore.
Reference model: doesn't have interface, but has writecore. 
My intention is to tell apart these two cases: to make interface of reference model null.

It's just when creating `ObjectTypeSerialization`, if we pass in `null` how could we tell of it is explicitly set or default `null`? https://github.com/Azure/autorest.csharp/blob/788cb700a0f236ac3adc8de6100bb848cb312d15/src/AutoRest.CSharp/Common/Output/Models/Serialization/ObjectTypeSerialization.cs#L29

Reply:
sounds like we should have another flag `hasWriteCore` instead of dealing with `Interfaces` differently